pgdrive/utils/basic_utils.py

```diff
@@ -132,3 +136,27 @@ def get_all_lanes(roadnet: "RoadNetwork"):
             for l in lanes:
                 res.append(l)
     return res
+
+
+def ray_localization(position: np.ndarray, pg_world: PgWorld) -> Tuple:
+    """
+    Get the index of the lane closest to a physx_world position.
+    Only used when smoething is on lane ! Otherwise fall back to use get_closest_lane()
+    :param position: a physx_world position [m].
+    :param pg_world: PgWorld class
+    :return: the index of the closest lane.
+    """
+    results = pg_world.physics_world.rayTestAll(Vec3(position[0], -position[1], 1.0),
+                                                Vec3(position[0], -position[1], -1))
+    lane_index_dist = []
+    if results.hasHits():
+        for res in results.getHits():
+            if res.getNode().getName() == BodyName.Lane:
+                lane = res.getNode().getPythonTag(BodyName.Lane)
+                lane_index_dist.append((lane.info, lane.index, lane.info.distance(position)))
```

Comment:
You use `if res.getNode().getName() == BodyName.Lane:`

Is there any possible to hit other kind of node here?

Reply:
yes, terrain node will also be hit. Use this to filter and get all overlap lanes